docs/source/api/data-management/backend-primitives.rst

```diff
@@ -0,0 +1,39 @@
+.. ******************************************************************************
+.. * Copyright contributors to the oneDAL project
+.. *
+.. * Licensed under the Apache License, Version 2.0 (the "License");
+.. * you may not use this file except in compliance with the License.
+.. * You may obtain a copy of the License at
+.. *
+.. *     http://www.apache.org/licenses/LICENSE-2.0
+.. *
+.. * Unless required by applicable law or agreed to in writing, software
+.. * distributed under the License is distributed on an "AS IS" BASIS,
+.. * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. * See the License for the specific language governing permissions and
+.. * limitations under the License.
+.. *******************************************************************************/
+
+.. highlight:: cpp
+
+    .. _api_dm_backend_primitives:
+
+==================
+Backend Primitives
+==================
+
+Refer to :ref:`Developer Guide: Backend Primitives <dm_backend_primitives>`.
+
+.. _backend_primitives_programming_interface:
+
+---------------------
+Programming interface
+---------------------
+
+All types and functions in this section are declared in the
+``oneapi::dal::backend::primitives`` namespace and be available via inclusion of the
+``oneapi/dal/backend/primitives/ndarray.hpp`` header file.
+
+.. toctree::
+
+   backend/ndarray.rst
```

Comment:
What's the relationship between "tables" and "arrays"?

Reply:
Good point. Will an example on how tables and ndarrays/ndviews are used together and the description of the differences.

Short answer:
- tables are user-facing API, always 2-dimensional.
- ndarray is internal convenience API, not available in release headers. Multidimensional.